### PR TITLE
feat: add ability to set VectorOfArray with Array using broadcast

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/vector_of_array.jl [hunk 1]
```diff
@@ -663,7 +663,20 @@ end
...
+        else
+            dest[:, i] = copy(unpack_voa(bc, i))
+        end
+    end
+    dest
+end
+
+@inline function Base.copyto!(dest::AbstractVectorOfArray,
+    bc::Broadcast.Broadcasted{<:Broadcast.DefaultArrayStyle})
+    bc = Broadcast.flatten(bc)
```
Comment (ChrisRackauckas) [inline src/vector_of_array.jl:677]: won't this allocate?

Comment (ChrisRackauckas) [reply src/vector_of_array.jl:677]: guess not, I'm adding a test for that.

File: test/interface_tests.jl [hunk 2]
```diff
@@ -125,3 +125,11 @@ z = VectorOfArray([zeros(SVector{2, Float64}), zeros(SVector{2, Float64})])
 z .= x .+ y
 
 @test z == VectorOfArray([fill(4, SVector{2, Float64}), fill(2, SVector{2, Float64})])
+
```
Comment (ChrisRackauckas) [inline test/interface_tests.jl:128]:
```suggestion

u1 = VectorOfArray([fill(2, SVector{2, Float64}), ones(SVector{2, Float64})])
u2 = VectorOfArray([fill(4, SVector{2, Float64}), 2 .* ones(SVector{2, Float64})])
u3 = VectorOfArray([fill(4, SVector{2, Float64}), 2 .* ones(SVector{2, Float64})])

function f(u1,u2,u3)
    u3 .= u1 .+ u2
end
f(u1,u2,u3)
@test@allocated f(u1,u2,u3) == 0 

```

File: src/vector_of_array.jl (hunk 1 above)
Comment (AayushSabharwal) [reply src/vector_of_array.jl:677]: Which part? `unpack_voa` creates views, so it won't if the target array is mutable. If the target isn't, then it has to allocate

File: test/interface_tests.jl [hunk 3]
```diff
@@ -125,3 +125,27 @@ z = VectorOfArray([zeros(SVector{2, Float64}), zeros(SVector{2, Float64})])
...
+u1 = VectorOfArray([fill(2, SVector{2, Float64}), ones(SVector{2, Float64})])
+u2 = VectorOfArray([fill(4, SVector{2, Float64}), 2 .* ones(SVector{2, Float64})])
+u3 = VectorOfArray([fill(4, SVector{2, Float64}), 2 .* ones(SVector{2, Float64})])
+
+function f(u1,u2,u3)
+    u3 .= u1 .+ u2
+end
+f(u1,u2,u3)
+@test @allocated f(u1,u2,u3) == 0
```
Comment (ChrisRackauckas) [inline test/interface_tests.jl:137]:
```suggestion
@test (@allocated f(u1,u2,u3)) == 0 
```

**Pull request merged**